File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        duplicateCount=0
        n=len(nums)
        for i in range(1,n):
            if nums[i]==nums[i-1]:
                duplicateCount+=1
            else:
                nums[i-duplicateCount]=nums[i]
        return n-duplicateCount

        

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int j=1;
        Set<Integer> s=new HashSet<>();
        s.add(nums[0]);
        for(int i=1; i < nums.length ; i++){
            if (s.contains(nums[i]))
                continue;
            nums[j]=nums[i];
            s.add(nums[j]);
            j++;           
           
        }
        return s.size();
        
    }
}